ssr/src/page/wallet/share_token_popup.rs

```diff
@@ -0,0 +1,171 @@
+use leptos::*;
+use leptos_icons::*;
+
+use crate::{component::overlay::*, utils::web::copy_to_clipboard};
+
+#[component]
+fn ShareProfileContent(
+    share_link: String,
+    message: String,
+    close_popup: RwSignal<bool>,
+) -> impl IntoView {
+    // let has_share_support = check_share_support();
+
+    let share_link_social = share_link.clone();
+
+    // Encode the message for URLs
+    let copy = share_link.clone();
+    let copy_clipboard = move |_| {
+        copy_to_clipboard(&copy);
+    };
+    view! {
+        <div class="flex flex-col gap-6 items-center p-6 w-full h-full bg-white rounded-lg shadow-lg">
+            <div class="flex flex-col gap-2 items-center">
+        <img class="w-16 h-16 md:w-20 md:h-20" src="/img/android-chrome-384x384.png" alt="YRAL Logo" />
+
+        <span class="text-xl font-semibold text-center md:text-2xl">
+            Share this app
+        </span>
+    </div>
+            <SocialShare message=message.clone() share_link=share_link_social.clone() />
+          <div class="flex overflow-x-auto justify-center items-center px-10 mx-1 space-x-2 w-full rounded-xl border-2 border-neutral-700 h-[2.5rem] md:h-[5rem]">
+        <span class="text-lg text-black md:text-xl truncate">
+                {&share_link.clone()}
+            </span>
+            <button on:click=copy_clipboard >
+                <Icon class="w-6 h-6 text-black cursor-pointer" icon=icondata::BiCopyRegular />
+            </button>
+        </div>
+        <button on:click=move|_| close_popup.set(true)
+        class="py-4 w-3/4 text-lg text-center text-white rounded-full bg-primary-600"
+    >
+    Back
+    </button>
+
+        </div>
+    }
+}
```

Comment:
Some of the components in this file looks reusable. Should be moved inside components folder?